nav2_costmap_2d/include/nav2_costmap_2d/costmap_topic_collision_checker.hpp

```diff
@@ -64,31 +56,29 @@ class CostmapTopicCollisionChecker
   /**
    * @brief Returns the obstacle footprint score for a particular pose
    */
-  double scorePose(const geometry_msgs::msg::Pose2D & pose, bool updateCostmap = true);
+  double scorePose(
+    const geometry_msgs::msg::Pose2D & pose,
+    bool fetch_costmap_and_footprint = true);
   /**
    * @brief Returns if a pose is collision free
    */
-  bool isCollisionFree(const geometry_msgs::msg::Pose2D & pose, bool updateCostmap = true);
+  bool isCollisionFree(
+    const geometry_msgs::msg::Pose2D & pose,
+    bool fetch_costmap_and_footprint = true);
 
 protected:
-  /**
-   * @brief Set a new footprint
-   */
-  void unorientFootprint(const Footprint & oriented_footprint, Footprint & reset_footprint);
   /**
    * @brief Get a footprint at a set pose
    */
-  Footprint getFootprint(const geometry_msgs::msg::Pose2D & pose);
+  Footprint getFootprint(const geometry_msgs::msg::Pose2D & pose, bool fetch);
```

Comment:
I think `fetch` should be removed here or needs to default to `true` like the others so default behavior gets the latest footprint. A comment in the collision checker regarding these `fetch` and `fetch_costmap_and_footprint` would be nice to tell users why this is there (so if looping over many poses, using the same costmap / footprint since the costmap specifically would incur a pretty gnarly copy if working on the global costmap).

Reply:
Basically, just want this context to be made clear to future readers